### PR TITLE
Implement copy, cut and paste for multiple selected elements

### PR DESCRIPTION
**Pull request by weeman1337**
copy

**:heavy_check_mark: Checklist**

- [x] A changeset describing the change and affected packages ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#changelog-and-versioning)).
- [ ] Added or updated documentation.
- [x] Tests for new functionality and regression tests for bug fixes.
- [x] Screenshots or videos attached (for UI changes).
- [x] All your commits have a `Signed-off-by` line in the message ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#dco)).

### CODE REVIEW

File: src/components/Shortcuts/ClipboardShortcuts/ClipboardShortcuts.test.tsx [hunk 1]
```diff
@@ -118,21 +124,24 @@ describe('<CopyAndPasteShortcuts>', () => {
...
+      'Hello World 1 Hello World 2',
+    );
     expect(clipboardData.setData).toBeCalledWith(
       'text/html',
       expect.stringContaining('<span data-meta='),
     );
 
-    expect(activeSlide.getActiveElementId()).toBe(undefined);
-    expect(activeSlide.getElement('element-1')).toBeUndefined();
+    expect(activeSlide.getActiveElementIds()).toHaveLength(0);
```
Comment (maheichyk) [inline src/components/Shortcuts/ClipboardShortcuts/ClipboardShortcuts.test.tsx:143]: Across tests it is common to check it in a bit more explicit way:
```suggestion
    expect(activeSlide.getActiveElementIds()).toEqual([]);
```

File: src/components/Shortcuts/ClipboardShortcuts/ClipboardShortcuts.tsx [hunk 2]
```diff
@@ -61,19 +62,23 @@ export function ClipboardShortcuts() {
         return;
       }
 
-      const activeElementId = slideInstance.getActiveElementId();
+      const activeElementIds = slideInstance.getActiveElementIds();
 
-      if (!activeElementId) {
+      if (!activeElementIds.length) {
```
Comment (maheichyk) [inline src/components/Shortcuts/ClipboardShortcuts/ClipboardShortcuts.tsx:67]: it is much common across code to use `length`:
```suggestion
      if (activeElementIds.length === 0) {
```

File: src/components/Shortcuts/ClipboardShortcuts/ClipboardShortcuts.tsx [hunk 3]
```diff
@@ -98,21 +103,25 @@ export function ClipboardShortcuts() {
...
```
Comment (maheichyk) [inline src/components/Shortcuts/ClipboardShortcuts/ClipboardShortcuts.tsx:108]:
```suggestion
      if (activeElementIds.length === 0) {
```

File: src/components/Shortcuts/ClipboardShortcuts/ClipboardShortcuts.tsx [hunk 4]
```diff
@@ -135,12 +144,9 @@ export function ClipboardShortcuts() {
       const content = readFromClipboardData(event.clipboardData);
       const { elements } = deserializeFromClipboard(content);
 
-      if (elements && elements.length > 0) {
-        for (const element of elements) {
-          slideInstance.addElement(element);
-
-          // TODO: When we support multiselect, we should select all added elements
-        }
+      if (elements?.length) {
```
Comment (maheichyk) [inline src/components/Shortcuts/ClipboardShortcuts/ClipboardShortcuts.tsx:147]: I think this statement was fine, no need to change.

File: src/state/types.ts [hunk 5]
```diff
@@ -189,7 +189,9 @@ export type WhiteboardSlideInstance = {
    * @deprecated to be replaced with getActiveElementIds
    * */
   getActiveElementId(): string | undefined;
-  /** Return the active elements */
+  /**
+   * @returns IDs of the currently selected elements in the order of their selection.
```
Comment (maheichyk) [inline src/state/types.ts:193]: We use active that has a context of selected in many places, so lets change a bit wording, but thanks for correcting the comment.
```suggestion
Return ids of the currently active elements in the order of their selection.
```

File: src/components/Shortcuts/ClipboardShortcuts/ClipboardShortcuts.tsx [hunk 6]
```diff
@@ -98,21 +103,25 @@ export function ClipboardShortcuts() {
...
         return;
       }
 
-      const activeElement = slideInstance.getElement(activeElementId);
+      const orderedActiveElementIds =
+        slideInstance.sortElementIds(activeElementIds);
+      const activeElements = slideInstance.getElements(orderedActiveElementIds);
 
-      if (!activeElement) {
+      if (isEmpty(activeElements)) {
```
Comment (maheichyk) [inline src/components/Shortcuts/ClipboardShortcuts/ClipboardShortcuts.tsx:116]: Please use `Object.keys`

File: src/components/Shortcuts/ClipboardShortcuts/ClipboardShortcuts.tsx [hunk 7]
```diff
@@ -61,19 +62,23 @@ export function ClipboardShortcuts() {
...
```
Comment (maheichyk) [inline src/components/Shortcuts/ClipboardShortcuts/ClipboardShortcuts.tsx:75]: `activeElements` have a type we can just use to check if it is empty without using external function
```suggestion
      if (Object.keys(activeElements).length === 0) {
```

File: src/components/Shortcuts/ClipboardShortcuts/ClipboardShortcuts.tsx [hunk 8]
```diff
@@ -98,21 +103,25 @@ export function ClipboardShortcuts() {
...
         return;
       }
 
-      const content = serializeToClipboard({ elements: [activeElement] });
+      const content = serializeToClipboard({
+        elements: Object.values(activeElements),
+      });
       writeIntoClipboardData(event.clipboardData, content);
-      slideInstance.removeElement(activeElementId);
+      slideInstance.removeElements(activeElementIds);
```
Comment (maheichyk) [inline src/components/Shortcuts/ClipboardShortcuts/ClipboardShortcuts.tsx:124]: This was a last usage of `removeElement` in the code. I think we don't need singular version anymore. Let's delete if from api and use `removeElements` in existing tests.

**Review by maheichyk — CHANGES_REQUESTED** [review]
Looks good, some small changes requested

File: src/components/Shortcuts/ClipboardShortcuts/ClipboardShortcuts.tsx (hunk 8 above)
Comment (weeman1337) [reply src/components/Shortcuts/ClipboardShortcuts/ClipboardShortcuts.tsx:124]: Done

File: src/components/Shortcuts/ClipboardShortcuts/ClipboardShortcuts.tsx (hunk 4 above)
Comment (maheichyk) [reply src/components/Shortcuts/ClipboardShortcuts/ClipboardShortcuts.tsx]: Sounds good to me

**Pull request merged**